src/setupTests.ts

```diff
@@ -3,3 +3,42 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import "@testing-library/jest-dom";
+
+import * as fs from "fs";
+import * as path from "path";
+import * as jsii from "@jsii/spec";
+import * as reflect from "jsii-reflect";
+
+// randomly selected...
+const ASSEMBLY_UNDER_TEST = "@aws-cdk/aws-ecr";
+
+function createAssembly(): reflect.Assembly {
+  const ts = new reflect.TypeSystem();
+
+  const packages = `${__dirname}/__assemblies__`;
+
+  collectAssebmlies(packages, ts);
+
+  return ts.findAssembly(ASSEMBLY_UNDER_TEST);
+}
+
+// expose the type system under test gloablly
+(global as any).assembly = createAssembly();
+
+function collectAssebmlies(p: string, ts: reflect.TypeSystem) {
+  const stat = fs.statSync(p);
+
+  if (stat.isDirectory()) {
+    fs.readdirSync(p).map((f) => collectAssebmlies(path.join(p, f), ts));
+  } else {
+    if (!p.endsWith("jsii.json")) {
+      throw new Error(
```

Comment:
Does it actually requires throwing an exception? Shouldn't we just log it and move on?

Reply:
Yeah i'll change.